### PR TITLE
Added docs on through associations

### PR DESCRIPTION
**Pull request by Globegitter**
Not ready to merge yet.

According to @mphasize through associations are supported by sails even though the Sails docs to not reflect that.

@dmarcelino @devinivy @particlebanana @tjwebb does anyone of you know all the details on all the keywords? I would love to get this example described clearly in plain english, since it confuses myself sometimes and I don't even know what all the attributes exactly achieve. So if any of you know the details and could chip in that would be great.

Edit: Example is taken from https://gist.github.com/mphasize/aeaff696132357797e0b

### CODE REVIEW

**Comment by dmarcelino** [comment]
@Globegitter that example *may* work but it's not "officially" supported, it's more of a hack/workaround than a solution. In other words I think it's dangerous to document that until we get official support (balderdashy/waterline#705) because the definition may (and likely will) change in [waterline v0.11](https://github.com/balderdashy/waterline/milestones).

I say it will likely change because if you look at [hasManyThrough.venue.fixture.js](https://github.com/balderdashy/waterline-adapter-tests/blob/master/interfaces/associations/support/fixtures/hasManyThrough.venue.fixture.js) you'll see the way the authors intended to define a through model is very different that hack/workaround.

**Comment by rachaelshaw** [comment]
@particlebanana thoughts?

**Comment by particlebanana** [comment]
Right now `through` isn't an actual `through` association like you would expect. It can only have two attributes. The entire purpose of it is so you can map out an existing many-to-many relationship in your database.

By default when you create a many-to-many relationship in Waterline you get a join table who's name is created automatically. Using `through` you can tell it to use a model that can have a `tableName` attribute for the join table.

**Comment by rachaelshaw** [comment]
Going to go ahead and close this, since it's no longer experimental and has [its own documentation page](https://sailsjs.com/documentation/concepts/models-and-orm/associations/through-associations) now. Thanks for taking a stab at this back in the day, @Globegitter!
